File: app/src/app/product/productService.js
/**
 * Created by Asus on 19/7/2559.
 */
(function() {
  'use strict'
  angular
    .module('app')
    .factory('productService',productService);

  /** @ngInject */
  function productService($resource){
    return $resource('/product/:id', { id: '@_id' }, {
      update: {
        method: 'PUT' // this method issues a PUT request
      }});

  }
})();

File: src/main/java/com/smartfarmh2/product/ProductService.java
package com.smartfarmh2.product;

import java.util.List;

/**
 * Created by Asus on 20/7/2559.
 */
public interface ProductService {
    Product create (Product product);
    Product update(Product product);
    void delete(Long id);
    Product getProduct(Long id);
    List<Product> list();
}
